### PR TITLE
feat(envVar): included environment variables in template renderer and gradle reads properties file

### PR DESCRIPTION
**Pull request by neubs-bsi**
- Added the possibilities to give a properties file via gradle that contains variable names and values that can be credentials. File is given via the gradle build file and added as a map to TemplateRenderer
- TemplateRenderer can now handle System environment variables in all frontends, either given when command is executed or already in system.
- Documentation was updated for each frontend
- Test were written
   - GradleTest: reads in properties file containing variable ANTENNATESTVARIABLE
   - AbstractAntennaFrontendTest: includes System variable and tests whether it is set or not and normal tests check if rendering is correct, otherwise tests would fail due to non-existing enum {ANTENNATESTVARIABLE}

### CODE REVIEW

File: assembly/cli/src/main/java/org/eclipse/sw360/antenna/frontend/cli/AntennaCLIFrontend.java [hunk 1]
```diff
@@ -41,6 +41,7 @@ public static void main(String[] args) {
             }
 
             AntennaCLIFrontend frontend = new AntennaCLIFrontend(pomFilePath.toFile());
+
```
Comment (maxhbr) [inline assembly/cli/src/main/java/org/eclipse/sw360/antenna/frontend/cli/AntennaCLIFrontend.java]: unnecessary change in unrelated file

File: core/frontend-stubs/cli-frontend-stub/src/main/java/org/eclipse/sw360/antenna/frontend/stub/cli/AntennaCLISettingsReader.java [hunk 2]
```diff
@@ -203,4 +209,22 @@ public ToolConfiguration readSettingsToToolConfiguration(XmlSettingsReader reade
...
+            prop.load(input);
+
+            return prop.entrySet()
+                    .stream()
+                    .collect(Collectors.toMap(
+                            p -> p.getKey().toString(),
+                            p -> p.getValue().toString()
+                    ));
+        } catch (IOException e) {
+            throw new IllegalArgumentException("IO exception when reading properties file: " + e.getMessage());
```
Comment (maxhbr) [inline core/frontend-stubs/cli-frontend-stub/src/main/java/org/eclipse/sw360/antenna/frontend/stub/cli/AntennaCLISettingsReader.java]: `IllegalArgumentException` is not correct here:

> Thrown to indicate that a method has been passed an illegal or inappropriate argument.

but the argument `propertiesFile` was valid and the function was not used incorrectly.

File: core/frontend-stubs/cli-frontend-stub/src/main/java/org/eclipse/sw360/antenna/frontend/stub/cli/AntennaCLISettingsReader.java [hunk 3]
```diff
@@ -203,4 +209,22 @@ public ToolConfiguration readSettingsToToolConfiguration(XmlSettingsReader reade
...
+
+    private Map<String, Object> mapProperties(File propertiesFile) {
+        try (InputStream input = new FileInputStream(propertiesFile)) {
+            Properties prop = new Properties();
+
+            prop.load(input);
+
+            return prop.entrySet()
+                    .stream()
+                    .collect(Collectors.toMap(
```
Comment (maxhbr) [inline core/frontend-stubs/cli-frontend-stub/src/main/java/org/eclipse/sw360/antenna/frontend/stub/cli/AntennaCLISettingsReader.java]: Wasn't there at some point a filter which only allowed uppercase keys?

File: core/frontend-stubs/cli-frontend-stub/src/main/java/org/eclipse/sw360/antenna/frontend/stub/cli/AntennaCLISettingsReader.java (hunk 2 above)
Comment (neubs-bsi) [reply core/frontend-stubs/cli-frontend-stub/src/main/java/org/eclipse/sw360/antenna/frontend/stub/cli/AntennaCLISettingsReader.java]: I don't know why this is shown as changed, but this is already like this:
https://github.com/eclipse/antenna/blob/fc457dd0829e0dfe5aad2bd151bafe1abadba8e1/core/frontend-stubs/cli-frontend-stub/src/main/java/org/eclipse/sw360/antenna/frontend/stub/cli/AntennaCLISettingsReader.java#L138

What do you propose otherwise?

File: core/frontend-stubs/cli-frontend-stub/src/main/java/org/eclipse/sw360/antenna/frontend/stub/cli/AntennaCLISettingsReader.java (hunk 3 above)
Comment (neubs-bsi) [reply core/frontend-stubs/cli-frontend-stub/src/main/java/org/eclipse/sw360/antenna/frontend/stub/cli/AntennaCLISettingsReader.java]: This is reading in the properties file. We did not set any restrictions for those variables.

The uppercase keys you are referring to are in the TemplateRenderer for the system environment variables

File: core/frontend-stubs/cli-frontend-stub/src/main/java/org/eclipse/sw360/antenna/frontend/stub/cli/AntennaCLISettingsReader.java (hunk 2 above)
Comment (maxhbr) [reply core/frontend-stubs/cli-frontend-stub/src/main/java/org/eclipse/sw360/antenna/frontend/stub/cli/AntennaCLISettingsReader.java]: Then this is also duplicated code from line 142ff

File: core/frontend-stubs/cli-frontend-stub/src/main/java/org/eclipse/sw360/antenna/frontend/stub/cli/AntennaCLISettingsReader.java (hunk 3 above)
Comment (maxhbr) [reply core/frontend-stubs/cli-frontend-stub/src/main/java/org/eclipse/sw360/antenna/frontend/stub/cli/AntennaCLISettingsReader.java]: oh, sure, right.

**Comment by blaumeiser-at-bosch** [comment]
Travis build broken, merge this, although there is a broken build
